### PR TITLE
[Search] Run CMK tests weekly

### LINKED ISSUE #16576: Only run Search + Key Vault (CMK) tests weekly
Based on https://github.com/Azure/azure-sdk-for-net/pull/16484#discussion_r517407677, once weekly tests are realized, we need only test CMK scenarios weekly most likely. The surface area doesn't change much, and it would help reduce costs without sacrificing coverage significantly.

### PR DESCRIPTION
**Pull request by ShivangiReja**
Customer-managed keys (CMKs) which require provisioning a Key Vault with the "basic" pricing tier, which adds cost over the free tier and because CMK support doesn't change much, we only want to do that weekly.

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/16576

### CODE REVIEW

File: sdk/search/platform-matrix.json
Comment (benbp) [inline sdk/search/platform-matrix.json:6]: Missing comma and whitespace normalization

```suggestion
{
    "matrix": {
        "$IMPORT": "eng/pipelines/templates/stages/platform-matrix.json",
        "ArmTemplateParameters": "@{ enableCmk = $false }"
    }
}
```

File: sdk/search/tests.yml
Comment (benbp) [inline sdk/search/tests.yml:9]: `=` and `\` characters in the query have to be escaped, see last bullet of [docs](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/README.md#replacemodifyappend-values).

```suggestion
        - ArmTemplateParameters=.*/@{ enableCmk \= $true }
```

File: sdk/search/platform-matrix.json
Comment (benbp) [inline sdk/search/platform-matrix.json:2]: This should make the test job titles a little cleaner:

```suggestion
{
    "displayNames": {
        "@{ enableCmk = $false }": "",
        "@{ enableCmk = $true }": "CMK",
    },
    "matrix": {
```

File: sdk/search/tests.yml
Comment (benbp) [inline sdk/search/tests.yml:12]:
```suggestion
    MaxParallel: 2
    MatrixConfigs:
      - Name: Search_cmk
        Path: sdk/search/platform-matrix.json
        Selection: sparse
        GenerateVMJobs: true
```

File: sdk/search/test-resources.json
Comment (heaths) [inline sdk/search/test-resources.json:216]: Just to make sure: `SEARCH_KEYVAULT_URL` is already treated as an optional recorded variable and code is conditioned suitably? I suspect since you're just changing the condition i.e., it was already conditional.

**Review by heaths — APPROVED** [review]

File: sdk/search/test-resources.json
Comment (benbp) [inline sdk/search/test-resources.json:67]:
```suggestion
            "defaultValue": "[if(equals(parameter('enableCmk'), true)), 'basic', 'free')]",
```

File: sdk/search/test-resources.json
Comment (ShivangiReja) [reply sdk/search/test-resources.json:216]: Yes, "SEARCH_KEYVAULT_URL" is an optional recorded variable [here](https://github.com/Azure/azure-sdk-for-net/blob/03293ed1a353b3db03f646a9aa42de6821048384/sdk/search/Azure.Search.Documents/tests/Utilities/SearchTestEnvironment.cs#L86) and the code is also conditioned correctly [here](https://github.com/Azure/azure-sdk-for-net/blob/03293ed1a353b3db03f646a9aa42de6821048384/sdk/search/Azure.Search.Documents/tests/Samples/Sample06_EncryptedIndex.cs#L31-L34).

**Review by benbp — APPROVED** [review]
One syntax comment to get the arm template working, otherwise LGTM assuming the tests pass and are verifiably deploying the keyvault.

**Review by jsquire — APPROVED** [review]

File: sdk/search/test-resources.json
Comment (ShivangiReja) [inline sdk/search/test-resources.json:67]: @heaths [Weekly tests](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2043470&view=results) are working fine but Nightly pipeline is failing at [Deploy test resources](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2044091&view=logs&j=ff3a09b3-827d-5654-2747-8202a4b0e892&t=0c3c1616-d3f4-570e-5882-d7fd6c8ea151) step with `Operation would exceed 'free' tier service quota. You are using 1 out of 1 'free' tier service quota` error.

Seems like we need to use the "basic" tier for Nightly runs as well where we don't run Key Vault + CMK tests. I'm not sure how the tests were running with "free" tier before [this](https://github.com/Azure/azure-sdk-for-net/pull/16484) PR in Nov 2022 because [here](https://github.com/MicrosoftDocs/azure-docs/blob/main/includes/azure-search-limits-per-subscription.md) I don't see Search RP has changed it's limit since then. I'm not sure what changed and there's no great way to verify anymore.

Does running the CMK tests all the time and deploying a keyvault cost us anything significant, especially if we have to use a "basic" SKU anyway? do we need to introduce all these complexity, if we can't use "free" SKU?

I see other languages are also using "basic" tier" - [Java](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/search/test-resources.json#L35), [JS](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/search/test-resources.json#L14)

Comment (ShivangiReja) [inline sdk/search/test-resources.json:67]: @heaths [Weekly tests](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2043470&view=results) are working fine but Nightly pipeline is failing at [Deploy test resources](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2044091&view=logs&j=ff3a09b3-827d-5654-2747-8202a4b0e892&t=0c3c1616-d3f4-570e-5882-d7fd6c8ea151) step with `Operation would exceed 'free' tier service quota. You are using 1 out of 1 'free' tier service quota` error.

Seems like we need to use the "basic" tier for Nightly runs as well where we don't run Key Vault + CMK tests. I'm not sure how the tests were running with "free" tier before [this](https://github.com/Azure/azure-sdk-for-net/pull/16484) PR in Nov 2020 because [here](https://github.com/MicrosoftDocs/azure-docs/blob/main/includes/azure-search-limits-per-subscription.md) I don't see Search RP has changed it's limit since then. I'm not sure what changed and there's no great way to verify anymore.

Comment (ShivangiReja) [reply sdk/search/test-resources.json:67]: Does running the CMK tests all the time and deploying a keyvault cost us anything significant, especially if we have to use a "basic" SKU anyway? do we need to introduce all these complexity, if we can't use "free" SKU?

I see other languages are also using "basic" tier" - [Java](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/search/test-resources.json#L35), [JS](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/search/test-resources.json#L14)

**Comment by ShivangiReja** [comment]
Seems like we need to use the "basic" tier for Nightly runs as well where we don't run Key Vault + CMK tests. If we need to use "basic" tier anyway, then I think there is no need to introduce all these complexity.

**Pull request closed without merging**
